kythe/rust/extractor/src/bin/extractor.rs

```diff
@@ -0,0 +1,230 @@
+use std::path::{Path, PathBuf};
+use tempdir::TempDir;
+use zip::{write::FileOptions, ZipWriter};
+
+fn main() -> Result<()> {
+    let config = cli::parse_arguments();
+
+    // Retrieve the SpawnInfo from the extra action file
+    let spawn_info = get_spawn_info(&config.extra_action_path)?;
+
+    // Create temporary directory and run the analysis
+    let tmp_dir = TempDir::new("rust_extractor")
+        .with_context(|| "Failed to make temporary directory".to_string())?;
+    let compiler_arguments: Vec<String> = spawn_info.get_argument().to_vec();
+    save_analysis::generate_save_analysis(
+        compiler_arguments.clone(),
+        PathBuf::new().join(tmp_dir.path()),
+    )?;
+
+    // Create the output kzip
+    let kzip_file = File::create(&config.output_path)
+        .with_context(|| format!("Failed to create kzip file at path {:?}", config.output_path))?;
+    let mut kzip = ZipWriter::new(kzip_file);
+
+    // Loop through each source file and insert into kzip
+    // Collect input files
+    let rust_source_files: Vec<String> = spawn_info
+        .get_input_file()
+        .to_vec()
+        .iter()
+        .filter(|file| file.ends_with(".rs"))
+        .map(String::from) // Map the &String to a new String
+        .collect();
+    let mut required_input: Vec<CompilationUnit_FileInput> = Vec::new();
+    for file_path in &rust_source_files {
+        kzip_add_required_input(file_path, &mut kzip, &mut required_input)?;
+    }
+
+    // Grab the build target's output path
+    let build_output_path: &String = spawn_info
+        .get_output_file()
+        .get(0)
+        .ok_or_else(|| anyhow!("Failed to get output file from spawn info"))?;
+
+    // Add save analysis to kzip
+    let save_analysis_path: String = analysis_path_string(&build_output_path, &tmp_dir.path())?;
+    kzip_add_required_input(&save_analysis_path, &mut kzip, &mut required_input)?;
+
+    // Create the IndexedCompilation and add it to the kzip
+    let indexed_compilation = create_indexed_compilation(
+        rust_source_files,
+        compiler_arguments,
+        build_output_path.to_string(),
+        required_input,
+    );
+    let mut indexed_compilation_bytes: Vec<u8> = Vec::new();
+    indexed_compilation
+        .write_to_vec(&mut indexed_compilation_bytes)
+        .with_context(|| "Failed to serialize IndexedCompilation to bytes".to_string())?;
+    let indexed_compilation_digest = sha256digest(&indexed_compilation_bytes);
+    kzip_add_file(
+        format!("pbunits/{}", indexed_compilation_digest),
+        &indexed_compilation_bytes,
+        &mut kzip,
+    )?;
+
+    println!("Generated {:?}", &config.output_path);
+    Ok(())
+}
+
+// Inspired by https://gist.github.com/Igosuki/27e18369fec5a05d4019ac5d321e1779
+const SPAWN_INFO: protobuf::ext::ExtFieldOptional<
+    ExtraActionInfo,
+    protobuf::types::ProtobufTypeMessage<SpawnInfo>,
+> = protobuf::ext::ExtFieldOptional { field_number: 1003, phantom: ::std::marker::PhantomData };
```

Comment:
Please use a named constant instead of `1003`, preferably somewhere from the generated protobuf code.

Reply:
The generated Rust code strips all field numbers, but I have created a constant variable stating that it is a field number